iohub/ngff/nodes.py

```diff
@@ -1144,57 +1145,71 @@ def set_transform(
         self.dump_meta()
 
     def set_scale(
-        self,
-        image: str | Literal["*"],
-        axis_name: str,
-        new_scale: float,
+        self, image: str | Literal["*"], axis_name: str, new_scale: float
     ):
         """Set the scale for a named axis.
         Either one image array or the whole FOV.
 
         Parameters
         ----------
-        image : str | Literal[
+        image : str | Literal['*']
             Name of one image array (e.g. "0") to transform,
             or "*" for the whole FOV
         axis_name : str
             Name of the axis to set.
         new_scale : float
             Value of the new scale.
         """
-        if len(self.metadata.multiscales) > 1:
-            raise NotImplementedError(
-                "Cannot set scale for multi-resolution images."
-            )
-
         if new_scale <= 0:
-            raise ValueError("New scale must be positive.")
-
+            raise ValueError(
+                f"New scale {axis_name}: {new_scale} is not positive!"
+            )
+        if image not in self and image != "*":
+            raise KeyError(f"Image {image} not found.")
         axis_index = self.get_axis_index(axis_name)
-
+        # Update scale while preserving existing transforms
+        if image == "*":
+            transforms = (
+                self.metadata.multiscales[0].coordinate_transformations or []
+            )
+        else:
+            for dataset_meta in self.metadata.multiscales[0].datasets:
+                if dataset_meta.path == image:
+                    transforms = dataset_meta.coordinate_transformations
+                    break
         # Append old scale to metadata
         iohub_dict = {}
         if "iohub" in self.zattrs:
             iohub_dict = self.zattrs["iohub"]
-        iohub_dict.update({f"prior_{axis_name}_scale": self.scale[axis_index]})
-        self.zattrs["iohub"] = iohub_dict
-
-        # Update scale while preserving existing transforms
-        transforms = (
-            self.metadata.multiscales[0].datasets[0].coordinate_transformations
+        if "previous_transforms" not in iohub_dict:
+            iohub_dict["previous_transforms"] = []
+        iohub_dict["previous_transforms"].append(
+            {
+                "image": image,
+                "transforms": [
+                    t.model_dump(**TO_DICT_SETTINGS) for t in transforms
+                ],
+                "modified": datetime.now().isoformat(),
```

Comment:
Is it worth keeping all of the previous transforms or just the most recent one? Flexible on this.

Reply:
I made it so that when the CLI modifies multiple axes in a loop it doesn't overwrite itself over and over again.